test/e2e/node_feature_discovery.go

```diff
@@ -485,6 +599,51 @@ func waitForNfdNodeLabels(cli clientset.Interface, expected map[string]string) e
 	return err
 }
 
+// waitForNfdNodeTaints waits for node to be tainted as expected.
+func waitForNfdNodeTaints(cli clientset.Interface, expectedTaints []corev1.Taint) error {
+	poll := func() error {
+		nodeList, err := cli.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, node := range nodeList.Items {
+			taints, err := nfdTaints(node.Spec.Taints, node)
+			if err != nil {
+				return fmt.Errorf("failed to fetch nfd owned taints for node: %s", node.Name)
+			}
+			if diff := cmp.Diff(taints, expectedTaints); diff != "" {
```

Comment:
You probably want to diff the other way around (sync with the error msg below)
```suggestion
			if diff := cmp.Diff(expectedTaints, taints); diff != "" {
```